client/examples/read_to_csv.py

```diff
@@ -17,44 +22,80 @@
             help='Maximum number of samples to record. Default 0 (no maximum)')
     parser.add_argument('-t','--timeout', type=int, default=0,  
             help='Collection time for sampling (s). Default is 0 (no timeout). The nominal sample rate is 5Hz.')
-    parser.add_argument('-p', '--port', default='/dev/ttyACM0',  
-            help='Serial port name. Default is /dev/ttyACM0.')
-    parser.add_argument('filename', help='CSV file for data output')
+    parser.add_argument('-p', '--ports', default='/dev/ttyACM0',  nargs='+',
+            help='Serial port name(s). Default is /dev/ttyACM0.')
+    # parser.add_argument('-f', '--filename', default='test.csv', help='CSV file for data output')
     args = parser.parse_args()
 
     logging.info("Starting demo")
     logging.info(args)
 
-    logging.info("Creating sink {}".format(args.filename))
-    csv = CsvSampleSink(args.filename)
-    csv.open()
-    
-    pt = board.Controller(port=args.port, sinks=[csv])
-    logging.info("Board ID: {}".format(pt.board_id()))
+    for index, item in enumerate(args.ports):
+        # print(index)
+        # print(item)
+        logging.info("Creating sink for {}".format(item))
+    # logging.info("Creating sink {}".format(args.filename))
+        sinks.append(CsvSampleSink("{}-{}.csv".format(item, datetime.now().strftime("%Y-%m-%d_%Hh-%Mm-%Ss"))))
```

Comment:
I think this will fail on Linux where the serial port is e.g.  '/dev/ttyACM0', which will result in an invalid filename

Reply:
True, this edition is specific to Windows' serial port naming convention. Any suggestions? Seems obnoxious to have to parse the file more to resolve invalid file names, though there's probably a library/module which does that.

Reply:
This should do it: keep the -f argument on the input with the default, which is equivalent to `filename`

```
import datetime
import os.path
filename = '/some/path/test.csv'
(prefix, ext) = os.path.splitext(filename)

serial_ports = ['/dev/ttyACM0', 'COM1', '/dev/ttyACM1', 'COM2']

for port in serial_ports:
    port_label = port.split('/')[-1]
    sink_name = '_'.join((prefix,port_label,datetime.datetime.now().strftime('%Y%m%d_%H%M%S'))) + ext
    print(sink_name)
```

Reply:
You can test it if you have different port names on Windows